Packages/com.unity.inputsystem/InputSystem/Controls/AxisControl.cs

```diff
@@ -277,5 +304,28 @@ public override unsafe float EvaluateMagnitude(void* statePtr)
 
             return NormalizeProcessor.Normalize(value, min, max, 0);
         }
+
+        public OptimizedControlDataType CalculateOptimizedControlDataType()
+        {
+            var noProcessingNeeded =
+                clamp == Clamp.None &&
+                invert == false &&
+                normalize == false &&
+                scale == false;
+
+            if (noProcessingNeeded &&
+                m_StateBlock.format == InputStateBlock.FormatFloat &&
+                m_StateBlock.sizeInBits == 32 &&
+                m_StateBlock.bitOffset == 0)
+                return OptimizedControlDataType.Float;
+            if (noProcessingNeeded &&
+                     m_StateBlock.format == InputStateBlock.FormatBit &&
+                     // has to be 8, otherwise we might be mapping to a state which only represents first bit in the byte, while other bits might map to some other controls
+                     // like in the mouse where LMB/RMB map to the same byte, just LMB maps to first bit and RMB maps to second bit
+                     m_StateBlock.sizeInBits == 8 &&
+                     m_StateBlock.bitOffset == 0)
+                return OptimizedControlDataType.BoolAsByte;
+            return OptimizedControlDataType.NotOptimized;
```

Comment:
I think we might also need to check if the control uses state from another control (usesStateFromOtherControl) and make sure the other control also has non-default processing if so, potentially following a chain of state from other controls. This is getting messy

Reply:
Do we? I though if we use state from other control we just do that .. use the state memory.
Is there an example where one control uses state from another control and when reading first control we run processor stack from second control?